hudi-spark-datasource/hudi-spark3.2plus-common/src/main/scala/org/apache/spark/sql/catalyst/plans/logcal/HoodieMetadataTableValuedFunction.scala

```diff
@@ -0,0 +1,30 @@
+package org.apache.spark.sql.catalyst.plans.logcal
+
+import org.apache.hudi.common.util.ValidationUtils.checkState
+import org.apache.spark.sql.AnalysisException
+import org.apache.spark.sql.catalyst.expressions.{Attribute, Expression}
+import org.apache.spark.sql.catalyst.plans.logical.LeafNode
+
+object HoodieMetadataTableValuedFunction {
+
+  val FUNC_NAME = "hudi_metadata";
+
+  def parseOptions(exprs: Seq[Expression], funcName: String): (String, Map[String, String]) = {
+    val args = exprs.map(_.eval().toString)
+    if (args.size != 1) {
+      throw new AnalysisException(s"Expect arguments (table_name or table_path) for function `$funcName`")
+    }
+
+    val identifier = args.head
+
+    (identifier, Map("hoodie.datasource.query.type" -> "snapshot"))
```

Comment:
Should be snapshot by default. Need to set incremental when users pass `as.of.instant` in the query.